storage/storage_image.go

```diff
@@ -177,18 +177,18 @@ func (s *storageImageSource) GetManifest(instanceDigest *digest.Digest) (manifes
 
 // LayerInfosForCopy() returns the list of layer blobs that make up the root filesystem of
 // the image, after they've been decompressed.
-func (s *storageImageSource) LayerInfosForCopy() []types.BlobInfo {
+func (s *storageImageSource) LayerInfosForCopy() ([]types.BlobInfo, error) {
```

Comment:
At least some of the errors in here should probably cause `copy.Image` to fail.

Cc: @nalind

Reply:
I preferred to not touch this part as I am not really familiar with it.  I can change it if it is required for this PR

Reply:
Generally, if `storageImageSource.LayerInfosForCopy()` is returning `nil`, the copy attempt is going to fail later on when the digests don't match, since I think the majority of images use compressed layers.  So yeah, pretty much anywhere we return `nil` from there, we should start returning the error value that caused us to do so.

Reply:
I've added a patch that changes `LayerInfosForCopy` in `storage/storage_image.go` to propagate the errors.